SwiftCompilerSources/Sources/SIL/Function.swift

```diff
@@ -26,6 +26,10 @@ final public class Function : CustomStringConvertible, HasShortDescription, Hash
     return Location(bridged: bridged.getLocation())
   }
 
+  public var sourceFile: BridgedNullableSourceFile {
```

Comment:
Please don't expose any bridged data structures in APIs. In case `sourceFile` is only used as is and no members of it are used, it's fine to just add a `typealias SourceFile = BridgedSourceFile`.

Also:
* this API should return an optional SourceFile: `var sourceFile: SourceFile?`
* please implement this as a member of `DeclRef` and add `var Function.declRef: DeclRef?`

Reply:
> it's fine to just add a `typealias SourceFile = BridgedSourceFile`

Added, thanks

> * this API should return an optional SourceFile: `var sourceFile: SourceFile?`

Fixed, thanks

* please implement this as a member of `DeclRef` and add `var Function.declRef: DeclRef?`

I'm not sure if it's feasible - the underlying function is a member of `SILFunction` and not of `SILDeclRef` (see code below). Could you please clarify what do you mean when suggest to implement this as a member of `DeclRef`?

https://github.com/swiftlang/swift/blob/e798e9c3cc77cf41991782b39a97daa6135a7ffd/include/swift/SIL/SILFunction.h#L1494-L1495